packages/@aws-cdk/aws-servicecatalog/lib/cloudformation-template.ts

```diff
@@ -34,6 +35,21 @@ export abstract class CloudFormationTemplate {
     return new CloudFormationProductStackTemplate(productStack);
   }
 
+  /**
+   * Creates a product with the resources defined in the given product stack and retains all previously deployed product stack versions.
+   */
+  public static fromProductStackHistory(productStack: ProductStack, locked: boolean, directory?: string ): CloudFormationTemplate {
+    return new CloudFormationProductStackTemplate(productStack, locked, directory);
+  }
+
+  /**
+   * Creates a product from a previously deployed product stack snapshot.
+   */
+  public static fromProductStackSnapshot(productStack: ProductStack, directory?: string ): CloudFormationTemplate {
+    return new CloudFormationProductStackSnapshotTemplate(productStack, directory);
+  }
```

Comment:
I think both of these are not necessary anymore, as long as `productStackHistory.versionFromSnapshot('v1')` can return a subclass of `CloudFormationTemplate`.

Reply:
Okay I think we should be able to do that